sdk/python/feast/registry.py

```diff
@@ -69,54 +67,62 @@ def __init__(self, registry_path: str, repo_path: Path, cache_ttl: timedelta):
             raise Exception(
                 f"Registry path {registry_path} has unsupported scheme {uri.scheme}. Supported schemes are file and gs."
             )
-        self.cached_registry_proto_ttl = cache_ttl
+        # Read cache from the RegistryStore.
+        self.cached_registry_proto = self._get_registry_proto()
         return
 
     def _initialize_registry(self):
-        """Explicitly forces the initialization of a registry"""
-        self._registry_store.update_registry_proto(updater=None)
+        """Explicitly initializes the registry with an empty proto."""
+        self.cached_registry_proto = RegistryProto()
+        self.cached_registry_proto.registry_schema_version = REGISTRY_SCHEMA_VERSION
+        self._registry_store.update_registry_proto(self.cached_registry_proto)
 
-    def apply_entity(self, entity: Entity, project: str):
+    def apply_entity(self, entity: Entity, project: str, flush: bool = True):
         """
         Registers a single entity with Feast
 
         Args:
             entity: Entity that will be registered
             project: Feast project that this entity belongs to
+            flush: True if the change should be persisted immediately
```

Comment:
Nit, this should be phrased more like `Whether the changes made to the registry should be immediately made to the registry or not.`

Reply:
Good point.